pkg/fleets/controller.go

```diff
@@ -410,27 +411,37 @@ func (c *Controller) rollingUpdateDeployment(fleet *agonesv1.Fleet, active *agon
 	if err != nil {
 		return 0, err
 	}
-	if err := c.rollingUpdateRest(fleet, rest); err != nil {
+	if err := c.rollingUpdateRest(fleet, active, rest); err != nil {
 		return 0, err
 	}
 	return replicas, nil
 }
 
+func getReadyReplicaCountForGameServerSets(gss []*agonesv1.GameServerSet) int32 {
+	totalReadyReplicas := int32(0)
+	for _, gss := range gss {
+		if gss != nil {
+			totalReadyReplicas += gss.Status.ReadyReplicas
+		}
+	}
+	return totalReadyReplicas
+}
+
 // rollingUpdateActive applies the rolling update to the active GameServerSet
 // and returns what its replica value should be set to
 func (c *Controller) rollingUpdateActive(fleet *agonesv1.Fleet, active *agonesv1.GameServerSet, rest []*agonesv1.GameServerSet) (int32, error) {
 	replicas := active.Spec.Replicas
 	// always leave room for Allocated GameServers
 	sumAllocated := agonesv1.SumStatusAllocatedReplicas(rest)
 
-	// if the active spec replicas are greater than or equal the fleet spec replicas, then we don't
-	// need to another rolling update upwards.
-	// Likewise if the active spec replicas don't equal the active status replicas, this means we are
+	// if the active spec replicas don't equal the active status replicas, this means we are
 	// in the middle of a rolling update, and should wait for it to complete.
-
 	if active.Spec.Replicas != active.Status.Replicas {
 		return replicas, nil
 	}
+
+	// if the active spec replicas are greater than or equal the fleet spec replicas, then we don't
+	// need to another rolling update upwards.
```

Comment:
"then we don't need to another"? Not quite sure what you mean here.

Reply:
to do another rolling update, fixing this.